File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int start = 0,end = nums.length-1;
        int ind = -1;
        while(start<=end){
            int mid = (end+start)/2;
            if(nums[mid]==target)
                return mid;
            else if(nums[mid]<target){
                ind = mid;
                start = mid+1;
            }else{
                end = mid-1;
            }
        }
        return ind+1;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        ind,start,end = -1,0,len(nums)-1
        while start<=end:
            mid = (start+end)//2
            if nums[mid]==target:
                return mid
            elif nums[mid]>target:
                end = mid-1
            else:
                ind = mid
                start = mid+1
        return ind+1